tasks/main.yml

```diff
@@ -17,26 +12,328 @@
+      command: head /var/opt/mssql/log/errorlog
+      register: __mssql_errorlog_default_content
+      changed_when: false
+      when: "'errorlogfile' not in __mssql_errorlogfile.stdout"
+
+    - name: Ensure that the mssql-server service is stopped  # noqa 102
+      service:
+        name: mssql-server
+        state: stopped
+      notify: Restart the mssql-server service
+      when: >
+        ((__mssql_errorlog_custom_content.skipped is not defined) and
+        ("{{ mssql_edition | lower }} edition" not in
+        __mssql_errorlog_custom_content.stdout | lower)) or
+        ((__mssql_errorlog_default_content.skipped is not defined) and
+        ("{{ mssql_edition | lower }} edition" not in
+        __mssql_errorlog_default_content.stdout | lower))
+      # Skipping rule because {{}} are required around vars here
+
+    - name: Change the edition of MSSQL Server  # noqa 102
+      command: /opt/mssql/bin/mssql-conf set-edition
+      environment:
+        MSSQL_PID: "{{ mssql_edition }}"
+      register: __mssql_conf_set_edition
+      changed_when: '"The new edition is" in __mssql_conf_set_edition.stdout'
+      when: >
+        ((__mssql_errorlog_custom_content.skipped is not defined) and
+        ("{{ mssql_edition | lower }} edition" not in
+        __mssql_errorlog_custom_content.stdout | lower)) or
+        ((__mssql_errorlog_default_content.skipped is not defined) and
+        ("{{ mssql_edition | lower }} edition" not in
+        __mssql_errorlog_default_content.stdout | lower))
+      # Skipping rule because {{}} are required around vars here
+      notify: Restart the mssql-server service
+
+- name: Set the IP address setting for MSSQL Server
+  when:
+    - __mssql_conf_setup.skipped is defined
+    - mssql_ip_address is defined
+  block:
+    - name: Check if the IP address setting is defined
+      command: /opt/mssql/bin/mssql-conf get network ipaddress
+      ignore_errors: true
+      changed_when: false
+      register: __mssql_conf_get_ip_address
+
+    - name: Change the IP address setting
+      command: >
+        /opt/mssql/bin/mssql-conf set network.ipaddress {{ mssql_ip_address }}
+      when: >
+        ("No setting for the given" in __mssql_conf_get_ip_address.stdout) or
+        (__mssql_conf_get_ip_address.stdout | regex_search('[^: ]*$')
+        not in mssql_ip_address)
+      notify: Restart the mssql-server service
+
+- name: Set the new TCP port for MSSQL Server
+  command: "/opt/mssql/bin/mssql-conf set network.tcpport {{ mssql_tcp_port }}"
+  register: __mssql_conf_set_tcp_port
+  ignore_errors: true
+  changed_when: >
+    "SQL Server needs to be restarted in order to apply this setting."
+    in __mssql_conf_set_tcp_port.stdout
+  notify: Restart the mssql-server service
+  when:
+    - __mssql_conf_setup.skipped is defined
+    - mssql_tcp_port is defined
+
+- name: Input the {{ mssql_input_sql_file }} file to MSSQL Server
+  when: mssql_input_sql_file is defined
```

Comment:
note that when `mssql_input_sql_file is defined` this block of tasks is not idempotent - there is no way to know (I'm guessing) that the file has already been applied.

Reply:
I think this would be pretty complex to parse - and this is with a relatively simple sql input file - unless the `sqlcmd` can output a simple error code or boolean to say "something changed" or "nothing changed", I don't think we should try to make this idempotent by parsing the output of `sqlcmd`.  Let's wait and see what users say.